File: app/src/main/java/net/rmoreno/weatherapp/ui/HourlyActivity.java
package net.rmoreno.weatherapp.ui;

import android.app.Activity;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.Menu;
import android.view.MenuItem;

import net.rmoreno.weatherapp.R;
import net.rmoreno.weatherapp.adapters.HourlyAdapter;
import net.rmoreno.weatherapp.models.HourlyWeather;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;


public class HourlyActivity extends Activity {
    String ACTIVITY = "HOURLY ACTIVITY";

    RecyclerView mRecyclerView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_daily);

        mRecyclerView = (RecyclerView) findViewById(R.id.recycler_view);
        mRecyclerView.setLayoutManager(new LinearLayoutManager(this));

        Bundle extras = getIntent().getExtras();


        //converted to json
        String hourlyString = extras.getString("hello");
        try{

            ArrayList<HourlyWeather> hourlyWeatherArrayList = getHourlyWeatherData(hourlyString);
            setUIValues(hourlyWeatherArrayList);
        } catch(JSONException e){
            Log.d(ACTIVITY, "JSON EXCEPTION");
        }
    }



    public ArrayList<HourlyWeather> getHourlyWeatherData(String jsonData) throws JSONException {
        ArrayList<HourlyWeather> hourlyWeatherList = new ArrayList<>();

        JSONObject jsonObject = new JSONObject(jsonData);
        JSONObject hourly = jsonObject.getJSONObject("hourly");
        JSONArray data = hourly.getJSONArray("data");

        for (int i = 0; i < data.length(); i++) {
            HourlyWeather hourlyWeather = new HourlyWeather();

            hourlyWeather.setTime(data.getJSONObject(i).getLong("time"));
            hourlyWeather.setTemp(data.getJSONObject(i).getDouble("temperature"));
            hourlyWeather.setTimeZone(jsonObject.getString("timezone"));
            hourlyWeatherList.add(hourlyWeather);
        }

        return hourlyWeatherList;
    }

    public void setUIValues(ArrayList<HourlyWeather> hourly){

        //the hourly data is get inside the adapter class
        HourlyAdapter adapter = new HourlyAdapter(HourlyActivity.this, hourly);
        mRecyclerView.setAdapter(adapter);

    }
    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_daily, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();

        //noinspection SimplifiableIfStatement
        if (id == R.id.action_settings) {
            return true;
        }

        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/net/rmoreno/weatherapp/ui/HourlyActivity.kt
package net.rmoreno.weatherapp.ui

import android.app.Activity
import android.os.Bundle
import android.util.Log
import android.view.Menu
import android.view.MenuItem
import androidx.recyclerview.widget.LinearLayoutManager
import kotlinx.android.synthetic.main.activity_daily.*
import net.rmoreno.weatherapp.R
import net.rmoreno.weatherapp.adapters.HourlyAdapter
import net.rmoreno.weatherapp.models.HourlyWeather
import org.json.JSONException
import org.json.JSONObject
import java.util.*

class HourlyActivity : Activity() {
    private var ACTIVITY = "HOURLY ACTIVITY"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_daily)

        recycler_view.layoutManager = LinearLayoutManager(this)

        val extras = intent.extras

        //converted to json
        val hourlyString = extras.getString("hello")
        try {

            val hourlyWeatherArrayList = getHourlyWeatherData(hourlyString)
            setUIValues(hourlyWeatherArrayList)
        } catch (e: JSONException) {
            Log.d(ACTIVITY, "JSON EXCEPTION")
        }
    }

    @Throws(JSONException::class)
    private fun getHourlyWeatherData(jsonData: String): ArrayList<HourlyWeather> {
        val hourlyWeatherList = ArrayList<HourlyWeather>()

        val jsonObject = JSONObject(jsonData)
        val hourly = jsonObject.getJSONObject("hourly")
        val data = hourly.getJSONArray("data")

        for (i in 0 until data.length()) {
            val hourlyWeather = HourlyWeather()

            hourlyWeather.time = data.getJSONObject(i).getLong("time")
            hourlyWeather.setTemp(data.getJSONObject(i).getDouble("temperature"))
            hourlyWeather.timeZone = jsonObject.getString("timezone")
            hourlyWeatherList.add(hourlyWeather)
        }

        return hourlyWeatherList
    }

    private fun setUIValues(hourly: ArrayList<HourlyWeather>) {
        //the hourly data is get inside the adapter class
        val adapter = HourlyAdapter(hourly)
        recycler_view.adapter = adapter

    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        // Inflate the menu; this adds items to the action bar if it is present.
        menuInflater.inflate(R.menu.menu_daily, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        val id = item.itemId

        if (id == R.id.action_settings) {
            return true
        }

        return super.onOptionsItemSelected(item)
    }
}
